src/components/Layout/ApplicationBody.jsx

```diff
@@ -77,27 +98,47 @@ const ApplicationBody = () => {
         });
     };
 
-    const updateCardData = (newCard) => {
-        setCardsList((prevState) => prevState.map(el => {
-            return el.id === newCard.id ? newCard : el;
-        }));
-    };
+    const updateCardData = useCallback((newCard) => {
+        setCardsList((prevState) => {
+            const newState = prevState.map(el => (el.id === newCard.id ? newCard : el));
+
+            const newIsSomeSelected = newState.findIndex(el => (el.isSelected === true)) !== -1;
+            if (newIsSomeSelected !== isSomeSelected) setIsSomeSelected(newIsSomeSelected);
+
+            return newState;
+        });
+    }, [isSomeSelected]);
+
+    const deleteSelected = (event) => {
+        if ((event.type === 'keydown' || event.type === 'keyup') && event.code === 'Space') event.preventDefault();
+        if (event.type === 'keyup' || (event.type === 'keydown' && event.code !== 'Space')) return;
```

Comment:
What is the reason of this code here?

Reply:
By browser's default pressing "Space" - scrolls down the list/page. While on focused element it shall trigger action - so scrolling need to be disabled, otherwise - page will be scrolled down along with performing the action.